enhancements/node-tuning/mixed-cpu-node-plugin.md

```diff
@@ -0,0 +1,439 @@
+
+## Motivation
+
+There is a growing interest from telco users in Kubernetes/OpenShift as a cloud management platform.
+This proposal is motivated by the desire of optimizing the platform's capabilities for telco users,
+such as cost reduction, power efficiency, higher pod density and sustainability.
+
+The telco users have new use cases and requirements that especially related to resource management of the platform. 
+In order to address some of those needs,
+Kubelet CPU manager was created to allow users requesting for exclusive, isolated cpus.
+
+But there are additional use cases that require more granular control of CPU management.
+This enhancement aims to solve the following use case:
+When the CPU manager allocates an exclusive set of CPUs for a given workloads, the workload shall still have the option 
+to specify and access a set of shared CPUs, whether those shared CPUs are shared among containers in the 
+same pod or containers in different ones.
+
+By giving the users the flexibility to use exclusive CPUs only when it is an absolute necessity, we help them utilize
+their existing compute resources in an optimal manner.
+
+### User Stories
+
+* As a developer who's building DPDK application,
+I want to run my application in a container,
+giving full exclusive access to cpu cores for the busy-loops polling threads,
+but having the light-weight tasks like configuration processing and log printing not taking full cores,
+in order to improve the application density and make more efficient use of the CPUs.
+For that, my application container needs access to both shared and exclusive CPUs.
+
+  
+### Goals 
+* Increase pod density, power efficiency, cost reduction and sustainability by optimizing CPUs resource management. 
+* Provide a special-purpose at a runtime-level solution for containers in a pod to have the ability
+to request and subsequently be allocated both exclusive and shared CPUs.
+* Enable the telco-specific use case with a self-contained change,
+    minimizing the changes to the platform
+
+### Non-Goals
+* Introducing a generic mechanism in the platform that does involve Kubelet and pod spec changes.
+
+## Proposal
+
+#### node-plugin implementation stages
+
+To minimize the moving part and keep this feature as stable as possible
+as an initial stage, the node-plugin would be implemented in CRI-O as a CRI-O hook
+(similar to the existing performance hooks).
+
+As a second stage, the plugin should be exported into a separate component, decoupled from CRI-O,
+and be running as a separate
+[NRI-plugin](https://github.com/containerd/nri).
+
+#### plugin workflow
+
+Container processes are restricted
+to given [cpuset](https://man7.org/linux/man-pages/man7/cpuset.7.html) by their corresponding cgroup definition.
+The kernel restricts the processes of a container to run at the CPU (core) ids specified under its
+cgroup.
+
+The plugin appends the shared cpus under the cgroup's `cpuset.cpus`.
```

Comment:
Can we add a header that this section describes details for cgroups v1?

Reply:
This is done for both V1 and V2

Reply:
`cpuset.cpus` changes are done for both V1 and V2 in similar fashion. 

The main difference is with the quota which I explained the differences in detail several lines later.